Source/lighting.h

```diff
@@ -74,10 +74,113 @@ void ChangeVisionXY(int id, Point position);
 void ProcessVisionList();
 void lighting_color_cycling();
 
+template <typename F>
+auto CrawlFlipsX(Displacement mirrored, F function) -> typename std::result_of<decltype(function)(Displacement)>::type
+{
+	const Displacement Flips[] = { mirrored.flipX(), mirrored };
+	for (auto displacement : Flips) {
+		auto ret = function(displacement);
+		if (ret)
+			return ret;
+	}
+	return {};
+}
+
+template <typename F>
+auto CrawlFlipsY(Displacement mirrored, F function) -> typename std::result_of<decltype(function)(Displacement)>::type
+{
+	const Displacement Flips[] = { mirrored, mirrored.flipY() };
+	for (auto displacement : Flips) {
+		auto ret = function(displacement);
+		if (ret)
+			return ret;
+	}
+	return {};
+}
+
+template <typename F>
+auto CrawlFlipsXY(Displacement mirrored, F function) -> typename std::result_of<decltype(function)(Displacement)>::type
+{
+	const Displacement Flips[] = { mirrored.flipX(), mirrored, mirrored.flipXY(), mirrored.flipY() };
+	for (auto displacement : Flips) {
+		auto ret = function(displacement);
+		if (ret)
+			return ret;
+	}
+	return {};
+}
+
+constexpr int MaxCrawlRadius = 18;
+
+/**
+ * CrawlTable specifies X- and Y-coordinate deltas from a missile target coordinate.
+ *
+ * n=4
+ *
+ *    y
+ *    ^
+ *    |  1
+ *    | 3#4
+ *    |  2
+ *    +-----> x
+ *
+ * n=16
+ *
+ *    y
+ *    ^
+ *    |  314
+ *    | B7 8C
+ *    | F # G
+ *    | D9 AE
+ *    |  526
+ *    +-------> x
+ */
+
+template <typename F>
+auto Crawl(unsigned radius, F function) -> typename std::result_of<decltype(function)(Displacement)>::type

```

Comment:
```suggestion
auto Crawl(unsigned radius, F function) -> typename std::invoke_result_t<decltype(function), Displacement>
```

`std::result_of<F>` was deprecated in C++17 and removed in C++20, this is why the msvc builds are failing. Need to switch to `std::invoke_result_t<F, Args...>`. Also should `#include <type_traits>`. Beyond that I like the approach, decent enough given we can't use coroutine based ranges :(